README.md

```diff
@@ -38,13 +39,78 @@ App::get('/hello-world') // Define Route
         }
     );
 
-App::setMode(App::MODE_TYPE_PRODUCTION); // Define Mode
+Http::setMode(Http::MODE_TYPE_PRODUCTION); // Define Mode
 
-$app        = new App('America/New_York');
-$request    = new Request();
-$response   = new Response();
+$http        = new Http(new Server(), 'America/New_York');
+$http->run();
+```
+
+### Server Adapters
+
+Library now supports server adapters and currently there are two servers implemented. You can use the PHP FPM server or the swoole server.
+
+**Use PHP FPM server**
+
+```php
+require_once __DIR__ . '/../../vendor/autoload.php';
+
+use Utopia\Http\Http;
+use Utopia\Http\Adapter\FPM\Request;
+use Utopia\Http\Adapter\FPM\Response;
+use Utopia\Http\Adapter\FPM\Server;
+
+Http::get('/hello-world') // Define Route
+    ->inject('request')
+    ->inject('response')
+    ->action(
+        function($request, $response) {
+            $response
+              ->addHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
+              ->addHeader('Expires', '0')
+              ->addHeader('Pragma', 'no-cache')
+              ->json(['Hello' => 'World']);
+        }
+    );
+
+Http::setMode(Http::MODE_TYPE_PRODUCTION); // Define Mode
+
+$http        = new Http(new Server(), 'America/New_York');
+$http->run(new Request(), new Response());
+```
+
+**Using swoole server**
+
+```php
+require_once __DIR__ . '/../../vendor/autoload.php';
+
+use Utopia\Http\Http;
+use Utopia\Http\Adapter\Swoole\Request;
+use Utopia\Http\Adapter\Swoole\Response;
+use Utopia\Http\Adapter\Swoole\Server;
+
+Http::get('/hello-world') // Define Route
+    ->inject('request')
+    ->inject('response')
+    ->action(
+        function($request, $response) {
+            $response
+              ->addHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
+              ->addHeader('Expires', '0')
+              ->addHeader('Pragma', 'no-cache')
+              ->json(['Hello' => 'World']);
+        }
+    );
+
+Http::setMode(Http::MODE_TYPE_PRODUCTION); // Define Mode
+
+$server = new Server('0.0.0.0', '80');
 
-$app->run($request, $response);
+$server->onRequest(function (Request $request, Response $response) use ($server) {
+    $http = new Http($server, 'UTC');
+    $http->run($request, $response);
+});
```

Comment:
Haven't we abstracted this part?
Might also be a good idea to have a small section with all the list of possible hooks, beside the regular routes (init, shutdown, onRequest, onStart).

Reply:
Ah yes, readme will need to be updated.